app/scripts/controllers/serviceInstance.js

```diff
@@ -94,8 +159,24 @@ angular.module('openshiftConsole')
             };
           });
 
-        $scope.$on('$destroy', function(){
-          DataService.unwatchAll(watches);
+        DataService.list(serviceClassesVersion, {}, function(serviceClasses) {
```

Comment:
We should be able to get just the one service class for this instance. (You might just need to rebase to pick up my changes for this.)